src/main/resources/org/jenkinsci/plugins/workflow/libs/Messages.properties

```diff
@@ -23,3 +23,4 @@
 LibraryDecorator.could_not_find_any_definition_of_librari=Could not find any definition of libraries {0}
 ResourceStep.library_resource_ambiguous_among_librari=Library resource {0} ambiguous among libraries {1}
 ResourceStep.no_such_library_resource_could_be_found_=No such library resource {0} could be found.
+ResourceStep.no_matching_resource_found_in_library=Library resource {0} not found in library {1}, but was present in other library
```

Comment:
nit: Maybe reword to: "Library resource {0} not found in library {1}, although it was present in libraries {2}" and pass `contents.keySet()` as the third argument.